tests/src/test/test_demoapp.py

```diff
@@ -0,0 +1,148 @@
+import os
+import random
+from datetime import datetime
+
+import pytest
+from faker import Faker
+from playwright.sync_api import Page, Playwright, expect
+
+from src.test.base import PreFlightCheck
+from src.pages import demoapp
+
+
+class Demoapp(PreFlightCheck):
+
+    def __init__(self):
+        self.url = PreFlightCheck.get_config("DEMOAPP_URL")
+        self.browser = None
+
+    def setup(self, playwright: Playwright) -> Page:
+        """ Login if no context """
+        if self.browser is None:
+            self.browser = playwright.firefox.launch_persistent_context(
+                user_data_dir=f"./test_demoapp_{datetime.timestamp(datetime.now())}",
+                headless=os.getenv("HEADLESS") in ("true", "True", "yes"))
+        page = self.browser.pages[0]
+
+        page.goto(self.url)
+
+        return page
+
+
+class DemoAppData:
+
+    class Owner:
+        def __init__(self, first_name: str, last_name: str, address: str, city: str, phone: str):
+            self.first_name = first_name
+            self.last_name = last_name
+            self.address = address
+            self.city = city
+            self.telephone = phone
+
+    class Pet:
+        def __init__(self, name: str, type: str, birth_date=datetime.date(datetime.now()).isoformat()):
+            self.name = name
+            self.birth_date = birth_date
+            self.type = type
+
+
+    def __init__(self):
+        faker = Faker()
+        self.owner = self.Owner(faker.name(), faker.name(), faker.address(), faker.city(), str(random.randint(1, 1000000)))
+        self.pet = self.Pet(faker.name(), random.sample(["cat", "dog", "bird", "hamster", "lizard", "snake"], 6)[0])
+
+
+subject = Demoapp()
+data = DemoAppData()
+
+
+@pytest.mark.base
+def test_owners_page(playwright: Playwright) -> None:
+    page = subject.setup(playwright)
+
+    demoapp.list_owners(page)
+
+    expect(page.get_by_role("link", name="George Franklin")).to_be_visible()
+    expect(page.get_by_role("link", name="Betty Davis")).to_be_visible()
+    expect(page.get_by_role("link", name="Eduardo Rodriquez")).to_be_visible()
+    expect(page.get_by_role("link", name="Harold Davis")).to_be_visible()
+    expect(page.get_by_role("link", name="Peter McTavish")).to_be_visible()
+
+    demoapp.list_owners(page, 2)
+
+    expect(page.get_by_role("link", name="Jean Coleman")).to_be_visible()
+    expect(page.get_by_role("link", name="Jeff Black")).to_be_visible()
+    expect(page.get_by_role("link", name="Maria Escobito")).to_be_visible()
+    expect(page.get_by_role("link", name="David Schroeder")).to_be_visible()
+    expect(page.get_by_role("link", name="Carlos Estaban")).to_be_visible()
+
+
+@pytest.mark.extended
```

Comment:
How/where is the extended test setup started / called?

Reply:
You can call them using the Option "-m" - for now the are not used in the docker setup.